### PR TITLE
T7434 test plan in boot docs

### PR DESCRIPTION
**Pull request by gctucker**
* add a field with the test plan name in new boot document entries in the database.
* clean up (syntax issues and docstrings)

### CODE REVIEW

File: app/utils/callback/lava.py [hunk 1]
```diff
@@ -68,6 +68,7 @@
     models.FASTBOOT_KEY: "platform.fastboot",
     models.INITRD_KEY: "job.initrd_url",
     models.BOARD_KEY: "platform.name",
+    models.PLAN_KEY: "test.plan",
```
Comment (loysollivier) [inline app/utils/callback/lava.py]: So this is a new metadata field passed by LAVA ? Could you elaborate on what this test.plan is ? Is it just a metadata field that should be in the LAVA job description or is it "created" by LAVA one doing some testing ?

Comment (gctucker) [reply app/utils/callback/lava.py]: This is already passed by LAVA in the job meta-data, it is in fact set when the LAVA job is submitted.  Typically, a kernelci.org job will have this set to "boot".  As we're adding more tests, other test plans such as "kselftest" are being used.

Comment (loysollivier) [reply app/utils/callback/lava.py]: OK thanks for the info

**Comment by gctucker** [comment]
Update: this version will create separate boot entries when different test plans are being used.  This is fine as long as builds aren't used with more than one test plan, otherwise several boot entries will be created and the front-end is not designed to show this information, resulting in duplicate boot data.

One way to resolve this is to consolidate "boots" and "tests" to simply treat "boot" as a test plan and stop making the distinction between the two.

**Comment by gctucker** [comment]
After some more discussion, we've decided to stop creating a boot entry for each test run and instead store the boot data directly into the test entry.  So this PR can be dropped.

I've created #35 with just the clean-up patch as that won't hurt.

**Pull request closed without merging**